src/include/graph/graph.h

```diff
@@ -45,6 +36,68 @@ struct GraphScanResult {
  */
 class Graph {
 public:
+    // Virtual Implementation of iterator functions
+    // Iterators can't really be virtual, but we can create an iterator (GraphResult::iterator)
+    // which uses a polymorphic object for its implementation
+    class IteratorImpl {
+    public:
+        virtual ~IteratorImpl() = default;
+    };
+
+    class Iterator {
+    public:
+        explicit Iterator(GraphScanState* scanState) : scanState{scanState} {}
+        DEFAULT_BOTH_MOVE(Iterator);
+        Iterator(const Iterator& other) = default;
+
+        using iterator_category = std::input_iterator_tag;
+        using difference_type = std::ptrdiff_t;
+        using value_type =
+            std::pair<std::span<const common::nodeID_t>, std::span<const common::relID_t>>;
+        using pointer = value_type;
+        using reference = value_type;
+
+        std::pair<std::span<const common::nodeID_t>, std::span<const common::nodeID_t>>
```

Comment:
```suggestion
        value_type
```
I think you can directly use `value_type` here.